### PR TITLE
net: lib: nrf_cloud_rest: add shadow transform request function

### PR DESCRIPTION
**Pull request by jayteemo**
Add function nrf_cloud_rest_shadow_transform_request to the nrf_cloud_rest library.
Add demo of new function using the nrf_cloud_rest_fota sample.

IRIS-9188

### CODE REVIEW

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_rest.c [hunk 1]
```diff
@@ -73,8 +73,10 @@ LOG_MODULE_REGISTER(nrf_cloud_rest, CONFIG_NRF_CLOUD_REST_LOG_LEVEL);
 #define API_GET_PGPS_BASE		API_VER API_LOCATION "/pgps"
 
 #define API_DEVICES_BASE		"/devices"
-#define API_DEVICES_STATE_TEMPLATE	API_VER API_DEVICES_BASE "/%s/state"
-#define API_DEVICES_MSGS_TEMPLATE	API_VER API_DEVICES_BASE "/%s/messages"
+#define API_DEVICES_TEMPLATE		API_VER API_DEVICES_BASE "/%s"
+#define API_DEVICES_TRANSFORM_TEMPLATE	API_DEVICES_TEMPLATE "?transform=%s"
+#define API_DEVICES_STATE_TEMPLATE	API_DEVICES_TEMPLATE "/state"
```
Comment (plskeggs) [inline subsys/net/lib/nrf_cloud/src/nrf_cloud_rest.c:78]: What about other places in this file that use API_DEVICES_STATE_TEMPLATE and API_DEVICES_MSGS_TEMPLATE that expect there to be a `%s`?

Comment (jayteemo) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud_rest.c:78]: it is included in `API_DEVICES_TEMPLATE` which is: `"/devices/%s"`

Comment (plskeggs) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud_rest.c:78]: right, missed that thanks

**Review by plskeggs — APPROVED** [review]

File: samples/cellular/nrf_cloud_rest_fota/README.rst [hunk 2]
```diff
@@ -62,6 +62,8 @@ Check and configure the following configuration options for the sample:
 
 CONFIG_REST_FOTA_JOB_CHECK_RATE_MIN - Update check rate
    This configuration option defines how often the sample checks for FOTA updates.
+   This value can be modified at runtime by adding/updating the ``"fotaInterval"`` item the desired config section in the device's shadow.
```
Comment (peknis) [inline samples/cellular/nrf_cloud_rest_fota/README.rst:65]:
```suggestion
   You can modify this value at runtime by adding or updating the ``"fotaInterval"`` item of the desired config section in the device's shadow.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

**Review by glarsennordic — APPROVED** [review]

**Pull request merged**